csrc/scheduler/registry.cpp

```diff
@@ -164,164 +164,105 @@ size_t SchedulerRuntimeInfo::getAlignmentSize(TensorView* tv) {
   return alignment_size;
 }
 
-bool SchedulerEntry::sameAs(const SchedulerEntry* other) {
-  return heuristic_ == other->heuristic_ && params_->sameAs(other->params_);
-}
-
 namespace {
 //! A Utility for checking both dynamic and static part of
 //!  can schedule
-template <typename SchedulerType>
-bool checkCanSchedule(
-    Fusion* fusion,
-    SchedulerRuntimeInfo& runtime_info,
-    HeuristicSummary* data_cache = nullptr) {
+bool checkCanSchedule(Fusion* fusion, HeuristicType heuristic_type) {
```

Comment:
Only left common can check sections in this function, removing the template to use polymorphism instead of type dispatch.